konduit-serving-python-new/src/main/resources/ai/konduit/serving/pydata.py

```diff
@@ -0,0 +1,81 @@
+try:
+    import PIL.Image
+    __PIL_INSTALLED__ = True
+except ImportError:
+    __PIL_INSTALLED__ = False
+import numpy as np
+
+
+class BoundingBox(object):
+    def __init__(self, cx, cy, height, width, label='', probability=1.):
+        self.cx = cx
+        self.cy = cy
+        self.height = height
+        self.width = width
+        self.label = label
+        self.probability = probability
+
+    @classmethod
+    def fromXY(cls, x1, y1, x2, y2, label='', probability=1.):
+        cx = 0.5 * (x1 + x2)
+        cy = 0.5 * (y1 + y2)
+        w = x2 - x1
+        h = y2 - y1
+        return cls(cx, cy, h, w, label, probability)
+
+    @classmethod
+    def fromCWH(cls, cx, cy, height, width, label='', probability=1.):
+        return cls(cx, cy, height, width, label, probability)
+
+
+class Data(object):
+    @staticmethod
+    def _check_type(item):
+        atomic_types = (str, int, float, bool, bytes, memoryview, bytearray, Data, BoundingBox, np.ndarray)
+        if __PIL_INSTALLED__:
+            atomic_types += (PIL.Image.Image,)
+        if isinstance(item, atomic_types):
+            pass
+        elif isinstance(item, (list, tuple)):
+            if not item:
+                raise Exception("Empty list not allowed.")
+            if len(set(map(type, item))) > 1:
+                raise Exception("All items in list should have same type.")
+            subitem = item[0]
+            if isinstance(subitem, Data):
+                raise Exception("Unsupported type in list: Data.")
```

Comment:
List<Data> is something we'll need to allow eventually, to match Java.
Same with List<List<...>> below.